File: AddressBook/AddressBook/Dtos/EmailLabel.cs
using AddressBook.Common.Paging;
using System.ComponentModel.DataAnnotations;

namespace AddressBook.Dtos
{
    public class EmailLabelRes
    {
        public int EmailLabelId { get; set; }
        public string? Label { get; set; }
    }

    public class EmailLabelReqEdit
    {
        [Required, MaxLength(20)]
        public string? Label { get; set; }
    }

    public class EmailLabelReqSearch : PagedReq
    {

    }
}

File: AddressBook/react-client/src/dtos/EmailLabel.ts
import Common from "../utility/Common";
import { PagedReq } from "./PagedReq";

export interface EmailLabelRes {
  emailLabelId?: string;
  label?: string;
}

export class EmailLabelReqEdit {
  label?: string = "";
}

export class EmailLabelReqSearch extends PagedReq {
  constructor(
    {
      pageNumber = 1,
      pageSize = Common.DEFAULT_PAGE_SIZE,
      orderBy = "",
      searchText = "",
    }: PagedReq,
    {}
  ) {
    super({
      pageNumber: pageNumber,
      pageSize: pageSize,
      orderBy: orderBy,
      searchText: searchText,
    });
  }
}